components/raftstore/src/store/peer.rs

```diff
@@ -2644,6 +2652,40 @@ where
         }
     }
 
+    pub fn handle_gen_snap_task<T: Transport>(&mut self, ctx: &mut PollContext<EK, ER, T>) {
+        if let Some(mut gen_task) = self.mut_store().take_gen_snap_task() {
+            // If the snapshot gen precheck feature is enabled, the leader needs
+            // to complete a precheck with the target follower before the
+            // snapshot generation.
+            if ctx.feature_gate.can_enable(SNAP_GEN_PRECHECK_FEATURE) {
+                let time_now = TiInstant::now();
+                // We should send out the precheck request if it hasn't been
+                // sent or if we have passed the scheduled time for it.
+                if gen_task.next_precheck_time.is_none()
+                    || time_now >= gen_task.next_precheck_time.unwrap()
+                {
+                    let to_peer = gen_task.to_peer.clone();
+                    self.send_snap_gen_precheck_request(ctx, &to_peer);
+
+                    // Calculate the next precheck time. A randomized time
+                    // interval generated within the defined range is used.
+                    let rand_interval = rand::thread_rng().gen_range(
+                        SNAP_GEN_PRECHECK_MIN_INTERVAL_MS..SNAP_GEN_PRECHECK_MAX_INTERVAL_MS,
+                    );
+                    gen_task.next_precheck_time =
+                        Some(time_now + Duration::from_millis(rand_interval));
```

Comment:
Why not use how many times to decide

Reply:
I was thinking that time is easier to reason about, but I agree that using the iteration count is also simple enough. Reverted to the iteration based approach. 

Currently, the iteration count between sending requests is between [1, 5]. Not sure if that's a reasonable range.